packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts

```diff
@@ -261,28 +261,14 @@ export default class MongoMemoryReplSet extends EventEmitter {
     return server;
   }
 
-  async _waitForPrimary(timeout: number = 30000): Promise<boolean> {
-    if (!this.admin) {
-      return false;
-    }
-    const replStatus: ReplStatusResultT = await this.admin.command({
-      serverStatus: 1,
-      metrics: 0,
-      locks: 0,
-    });
-    this.debug('   replStatus:', replStatus);
-    const hasPrimary = replStatus.repl.ismaster;
-    if (!hasPrimary) {
-      const restTimeout = timeout - 500;
-      if (restTimeout <= 0) {
-        throw new Error(`No PRIMARY elected yet. Timeout expired. Exiting...`);
-      }
-      this.debug(`No PRIMARY yet. Waiting... ${restTimeout}ms`);
-      return new Promise((resolve) =>
-        setTimeout(() => resolve(this._waitForPrimary(restTimeout)), 500)
-      );
-    }
+  async _waitForPrimary(): Promise<void> {
+    await Promise.race(
+      this.servers.map((server) => {
+        const instanceInfo: any = server.getInstanceInfo();
```

Comment:
Yep.. typescript was giving me grief there.. so skipped that initially..  Just pushed update without any.. There is maybe more elegant way to write it if I am confident that getInstanceInfo won't return false, but its been some time I was actively using typescript. So new if statement to check of there is instanceInfo seems to do the trick

Reply:
@jardakotesovec Well you could use this workaround : 
```javascript
const instanceInfo: MongoInstanceDataT | false  = server.getInstanceInfo();
```
I dont really like the syntax but it's correct, the problem here is that the function `getInstanceInfo` does two things at the same time : 
* it checks if the instance is up
* it also returns the instance info so the return type of the function is a bit weird `MongoInstanceDataT | false ` :/

 I would have done two functions, one to check if the instance is up and one to gather the instance info but I don't think it's really important anyway.

Reply:
@AJRdev 
```
const instanceInfo: MongoInstanceDataT | false  = server.getInstanceInfo();
```
I think thats equivalent to
```
const instanceInfo = server.getInstanceInfo();
```

As it gets inferred from getInstanceInfo() function.

It still requires to putting there if to ensure the false is not hapenning to make typescript happy

Reply:
Yes you will always need the if if the return type is not a mandatory object, since you are accessing its field `instanceInfo.instance` afterwards and it could be possibly equal to false here. 
It was more to indicate to code user the type of the object without having to go to the function but it's maybe not that important since you're not using the object a lot afterward..

Reply:
Thats up to vs code to indicate what type it is :-).